pkg/executor/util/utils.go

```diff
@@ -99,23 +100,49 @@ func NewNoopStorageProvider(
 }
 
 func NewStandardExecutorProvider(
-	dockerCacheCfg types.DockerCacheConfig,
+	cfg types2.ExecutorsConfig,
 	executorOptions StandardExecutorOptions,
 ) (executor.ExecutorProvider, error) {
-	dockerExecutor, err := docker.NewExecutor(executorOptions.DockerID, dockerCacheCfg)
-	if err != nil {
-		return nil, err
+	providers := make(map[string]executor.Executor)
+	if cfg.Enabled(types2.KindExecutorDocker) {
+		var dockerExecutor *docker.Executor
+		if cfg.HasConfig(types2.KindExecutorDocker) {
+			dockercfg, err := types2.DecodeProviderConfig[types2.Docker](cfg)
+			if err != nil {
+				return nil, err
+			}
+			dockerExecutor, err = docker.NewExecutor(executorOptions.DockerID, types.DockerCacheConfig{
+				Size:      dockercfg.ManifestCache.Size,
+				Duration:  types.Duration(dockercfg.ManifestCache.TTL),
+				Frequency: types.Duration(dockercfg.ManifestCache.Refresh),
+			})
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			var err error
+			dockerExecutor, err = docker.NewExecutor(executorOptions.DockerID, types.DockerCacheConfig{
+				Size:      1000,
+				Duration:  types.Duration(1 * time.Hour),
+				Frequency: types.Duration(1 * time.Hour),
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+		providers[models.EngineDocker] = dockerExecutor
```

Comment:
@wdbaruni this is an example of how we configure a provider iff its enabled and dependent on if it contains a configuration - if it does we use it, if it doesn't we use a default setting. Does this align with your vision of things?

Reply:
https://github.com/bacalhau-project/bacalhau/pull/4333#discussion_r1728017505